eAPI 1.6

jiripudil opened the pull request:
see https://github.com/csob/paymentgateway/wiki/eAPI-1.6 + Kdyby/CsobPaymentGateway#16

jiripudil committed:
eAPI 1.6 bump dependency

jiripudil committed:
eAPI 1.6: version configuration

jiripudil committed:
eAPI 1.6: oneclickPayment

fprochazka commented on composer.json:
```diff
@@ -52,6 +52,12 @@
+	"repositories": [
+		{
+			"type": "vcs",
+			"url": "https://github.com/jiripudil/CsobPaymentGateway"
```
You have push rights AFAIK, you could have created the branch directly in `Kdyby/CsobPaymentGateway` :) Next time ;)

fprochazka commented on src/UI/CsobControl.php:
```diff
@@ -146,6 +151,33 @@ public function pay()
+	 */
+	private function oneclickPayment(Csob\Payment $payment)
+	{
+		// Note: We recommend to call payment/status not earlier than 2-3 seconds after payment/oneclick/start call.
```
This node should rather be in the PhpDoc, as it can be viewed using QuickDocumentationReference shortcut in the IDE, without jumping to the method.

fprochazka commented on src/UI/CsobControl.php:
```diff
@@ -146,6 +151,33 @@ public function pay()
+
+		$initResponse = $this->client->paymentOneclickInit($payment);
+		$this->client->paymentOneclickStart($initResponse->getPayId());
+		sleep(3);
```
Imho the polling should be separate method (accepting just `payId`), that will be not called here, but rather the user will call it manully, or move this easily to some (RPC) rabbit worker, etc

fprochazka commented on .travis.yml:
```diff
@@ -33,7 +33,7 @@ before_install:
 
 install:
-  - travis_retry composer update --no-interaction --prefer-dist $COMPOSER_EXTRA_ARGS
+  - travis_retry composer update --no-interaction --prefer-source $COMPOSER_EXTRA_ARGS
```
this is also no longer needed :)

jiripudil committed:
eAPI 1.6: status polling moved to a separate method

pay() may now return response with pending status, in which case polling should be utilized

fprochazka commented on src/UI/CsobControl.php:
```diff
@@ -100,7 +100,13 @@ public function pay()
+					$response = $this->client->paymentRecurrent($payment);
+
+				} else {
+					$initResponse = $this->client->paymentOneclickInit($payment);
```
are you sure this isn't coliding with `$this->client->createPayment()` ? Seems weird you'd have to call both "inits".

jiripudil replied:
`createPayment` merely creates and sets up the `Payment` object, `paymentInit` is called separately in the [other branch of this condition](https://github.com/Kdyby/CsobPaygateNette/pull/5/files#diff-07614aa4ce077262be1c7f0b078f2111R130). So yeah, I am pretty sure :)

fprochazka replied:
Ok, makes sense :)

fprochazka commented:
:+1:

jiripudil commented:
- [x] bump the dependency once Kdyby/CsobPaymentGateway#16 is merged and released

fprochazka commented:
@jiripudil great work!
